src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -123,48 +132,75 @@ protected AbstractEntityAIBasic(@NotNull final J job)
-                 */
+          /*
+            Update chestbelt and nametag
+            Will be executed every time
+            and does not stop execution
+           */
           new AITarget(this::updateVisualState),
-                /*
-                 * If waitingForSomething returns true
-                 * stop execution to wait for it.
-                 * this keeps the current state
-                 * (returning null would not stop execution)
-                 */
+          /*
+            If waitingForSomething returns true
+            stop execution to wait for it.
+            this keeps the current state
+            (returning null would not stop execution)
+           */
           new AITarget(this::waitingForSomething, this::getState),
-                /*
-                 * Check if any items are needed.
-                 * If yes, transition to NEEDS_ITEM.
-                 * and wait for new items.
-                 */
-          new AITarget(() -> getState() == NEEDS_ITEM || this.getOwnBuilding()
-                                                           .hasWorkerOpenRequestsFiltered(worker.getCitizenData(), r -> !worker.getCitizenData().isRequestAsync(r.getToken()))
-                               || this.getOwnBuilding().hasCitizenCompletedRequests(worker.getCitizenData()), this::waitForRequests),
-                /*
-                 * Dumps inventory as long as needs be.
-                 * If inventory is dumped, execution continues
-                 * to resolve state.
-                 */
+          /*
+            Check if any items are needed.
+            If yes, transition to NEEDS_ITEM.
+            and wait for new items.
+           */
+          new AITarget(() -> {
+              return getState() == NEEDS_ITEM
+                       || this.getOwnBuilding()
+                            .hasWorkerOpenRequestsFiltered(
+                              worker.getCitizenData(),
+                              r -> !worker.getCitizenData().isRequestAsync(r.getToken()))
+                       || this.getOwnBuilding().hasCitizenCompletedRequests(worker.getCitizenData());
+          }, this::waitForRequests),
+          /*
+            Dumps inventory as long as needs be.
+            If inventory is dumped, execution continues
+            to resolve state.
+           */
           new AITarget(INVENTORY_FULL, this::dumpInventory),
-                /*
-                 * Check if inventory has to be dumped.
-                 */
+          /*
+            Check if inventory has to be dumped.
+           */
           new AITarget(this::inventoryNeedsDump, INVENTORY_FULL),
-          /**
+          /*
            * Reset to idle if no specific tool is needed.
            */
-          new AITarget(() -> getState() == NEEDS_TOOL && this.getOwnBuilding().getOpenRequestsOfType(worker.getCitizenData(), TypeToken.of(Tool.class)).isEmpty(), IDLE)
+          new AITarget(() -> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove useless curly braces around statement and then remove useless return keyword (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1602)